src/sagas/clients.js

```diff
@@ -24,34 +25,31 @@ export function* createSnapshot() {
   }
 }
 
-export function* exportGist() {
+export function* exportProject({payload: {exportType}}) {
   const state = yield select();
   const project = getCurrentProject(state);
   const user = state.get('user').toJS();
-  try {
-    const {html_url} = yield call(createGistFromProject, project, user);
-    yield put(gistExported(html_url));
-  } catch (e) {
-    yield put(gistExportError(e));
-  }
-}
 
-export function* exportRepo() {
-  const state = yield select();
-  const project = getCurrentProject(state);
-  const user = state.get('user').toJS();
   try {
-    const {html_url} = yield call(createRepoFromProject, project, user);
-    yield put(repoExported(html_url));
+    if (exportType === 'gist') {
+      const {html_url: url} = yield call(createGistFromProject, project, user);
+      yield put(projectExported(url, exportType));
```

Comment:
This line is the same across all three logical branches—might as well just write it once after the `if`/`else`?

Reply:
Yeah I agree - I tried it at the end and there are linting errors
`error  'url' is assigned a value but never used`
not sure why this is

Reply:
Ah, because [`const` and `let` are block-local](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let)! But you should be able to fix that by declaring `let url;` before the `if` statement.

Reply:
Okay cool! How would I do that with the object destructing? The below throws an error at `{html_url: url}` 

``` 
 let url;

  try {
    if (exportType === 'gist') {
      {html_url: url} = yield call(createGistFromProject, project, user);
    } else if (exportType === 'repo') {
      {html_url: url} = yield call(createRepoFromProject, project, user);
    } else if (exportType === 'classroom') {
      const snapshotKey = yield call(createProjectSnapshot, project);
      url = yield call(createShareToClassroomUrl, snapshotKey);
    }
    yield put(projectExported(url, exportType));
  } 
```

The below throws gives back the `url` as undefined.

``` 
 let url;

  try {
    if (exportType === 'gist') {
      let {html_url: url} = yield call(createGistFromProject, project, user);
    } else if (exportType === 'repo') {
      let {html_url: url} = yield call(createRepoFromProject, project, user);
    } else if (exportType === 'classroom') {
      const snapshotKey = yield call(createProjectSnapshot, project);
      let url = yield call(createShareToClassroomUrl, snapshotKey);
    }
    yield put(projectExported(url, exportType));
  } 
```

Reply:
Yeah it’s gross but you have to do:

```js
({html_url: url} = yield call(createGistFromProject, project, user));
```

As in just literally [put parentheses around the whole statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) 😿

Reply:
```
  let url;

  try {
    if (exportType === 'gist') {
      const {html_url: html_url} = yield call(createGistFromProject, project, user);
      url = html_url;
    } else if (exportType === 'repo') {
      const {html_url: html_url} = yield call(createRepoFromProject, project, user);
      url = html_url;
    } else if (exportType === 'classroom') {
      const snapshotKey = yield call(createProjectSnapshot, project);
      url = yield call(createShareToClassroomUrl, snapshotKey);
    }
    yield put(projectExported(url, exportType));
}
```

Reply:
I didn't see above! Cool!